coremltools/converters/tensorflow/test/test_tf_2x.py

```diff
@@ -11,6 +11,102 @@
 import pytest
 
 
+@unittest.skipUnless(HAS_TF_2, 'missing TensorFlow 2+.')
+class TestSingleOp(unittest.TestCase):
+    # In this class we test tensorflow 2.x op without using Keras API
+
+    def _test_coreml(self, model, input_dic=None, output_names=None):
+
+        # Get concrete function
+        concrete_func = model.__call__.get_concrete_function()
+
+        # Get function input
+        if input_dic == None:
+            input_dic = []
+            for input in concrete_func.inputs:
+                name = input.name.split(':')[0]
+                shape = input.shape
+                if shape == None or any([x is None for x in input.shape.as_list()]):
+                    raise ValueError("Please specify 'input_dic' for dynamic shape input.")
+                shape = input.shape.as_list()
+                input_dic.append((name, shape))
+        else:
+            if not isinstance(input_dic, list):
+                raise TypeError("'input_dic' should be [(str, tensor)] type.")
+
+        inputs = [(name, np.random.uniform(-1,1,shape).astype(np.float32), shape)
+                  for name, shape in input_dic]
+
+        # Get output names
+        if output_names == None:
+            output_names = [output.name.split(':')[0] for output in concrete_func.outputs]
+        else:
+            if not isinstance(output_names, list):
+                raise TypeError("'output_names' should be [str] type.")
+        # Tensorflow predict
+        tf_inputs = [tf.convert_to_tensor(value) for name, value, shape in inputs]
+        tf_outputs = model(*tf_inputs)
+
+        # Coreml model predict
```

Comment:
Here, I find that coremltools automatically convert the input shape with [] to [1,] hmmm.
So I write a workaround to make the control flow unit test test hmmm.
@1duo  @aseemw did you have more context about why this is the case?